### PR TITLE
[twentythreevideo] support subdomain urls

### PR DESCRIPTION
**Pull request by dancek**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

I came across bonnier-publications-danmark.23video.com urls and wanted to download them. I presume 23video supports custom subdomains for enterprise users, so I changed the code to accept any subdomain of the known domains.

If this PR is considered useful, I'd be really happy if you could tag it `hacktoberfest-accepted`.

### CODE REVIEW

**Comment by dancek** [comment]
The video I added in `_TESTS` can be found in https://etunurkka.digi-kuva.fi/kuvankasittely/photoshop-elements/4-photoshop-elements-2019n-uutuusominaisuutta btw.

File: youtube_dl/extractor/twentythreevideo.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
 class TwentyThreeVideoIE(InfoExtractor):
     IE_NAME = '23video'
-    _VALID_URL = r'https?://video\.(?P<domain>twentythree\.net|23video\.com|filmweb\.no)/v\.ihtml/player\.html\?(?P<query>.*?\bphoto(?:_|%5f)id=(?P<id>\d+).*)'
-    _TEST = {
+    _VALID_URL = r'https?://(?P<domain>[^.]+\.(twentythree\.net|23video\.com|filmweb\.no))/v\.ihtml/player\.html\?(?P<query>.*?\bphoto(?:_|%5f)id=(?P<id>\d+).*)'
```
Comment (dstftw) [inline youtube_dl/extractor/twentythreevideo.py:11]: Don't capture groups you don't use.

File: youtube_dl/extractor/twentythreevideo.py [hunk 2]
```diff
@@ -21,11 +21,23 @@ class TwentyThreeVideoIE(InfoExtractor):
...
+        'md5': '772a91f83d129ee5f015b12bea61a78b',
+        'info_dict': {
+            'id': '36137620',
+            'ext': 'mp4',
+            'upload_date': '20181004',
+            'uploader': 'Kristoffer Engbo',
+            'title': 'Photoshop Elements 2019 - Photo Text',
+            'uploader_id': '10801356',
+            'timestamp': 1538664032,
+        }
```
Comment (dstftw) [inline youtube_dl/extractor/twentythreevideo.py:35]: `only_matching`.

**Review by dstftw — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/twentythreevideo.py (hunk 1 above)
Comment (dancek) [reply youtube_dl/extractor/twentythreevideo.py:11]: @dstftw do you mean this should be written `(?P<domain>[^.]+\.twentythree\.net|[^.]+\.23video\.com|[^.]+\.filmweb\.no)`?

The named groups `domain`, `query` and `id` are used.

Comment (dstftw) [reply youtube_dl/extractor/twentythreevideo.py:11]: No.

Comment (dancek) [reply youtube_dl/extractor/twentythreevideo.py]: Sorry for not doing my homework, I only learned of non-capturing groups now.

**Comment by dancek** [comment]
Thanks for the review! I hope this fixed version is ok.

**Pull request merged**